SilKit/source/core/vasio/VAsioPeer.cpp

```diff
@@ -210,34 +217,29 @@ void VAsioPeer::DispatchBuffer()
     }
 
 
-    if (_wPos < _currentMsgSize)
+    if (_msgBuffer.Size() < _currentMsgSize)
     {
         // Make the buffer large enough and wait until we have more data.
-        _msgBuffer.resize(_currentMsgSize);
+        if (_msgBuffer.Capacity() < _currentMsgSize)
+        {
+            _msgBuffer.Reserve(_currentMsgSize);
+        }
+
         ReadSomeAsync();
         return;
     }
     else
     {
-        auto newBuffer = std::vector<uint8_t>{_msgBuffer.begin() + _currentMsgSize, _msgBuffer.end()};
-        auto newWPos = _wPos - _currentMsgSize;
-
-        // manually extract the message size so we can adjust the MessageBuffer size
-        uint32_t msgSize{0u};
-        if (_msgBuffer.size() < sizeof msgSize)
+        std::vector<uint8_t> currentMsg(_currentMsgSize);
+        if (!_msgBuffer.Read(currentMsg))
         {
-            throw SilKitError{"DispatchBuffer: Received message is too small to contain message size header"};
+            throw SilKitError("Reading data from ring buffer failed.");
         }
-        memcpy(&msgSize, _msgBuffer.data(), sizeof msgSize);
-        //ensure buffer does not contain data from contiguous messages
-        _msgBuffer.resize(msgSize);
-        SerializedMessage message{std::move(_msgBuffer)};
+
+        SerializedMessage message{std::move(currentMsg)};
```

Comment:
We could maybe save one more copy if we refactor SerializedMessage to work on a BufArray (like a vector view). We could advance Rpos after SerializedMessage consumes the date in the BufArray

Reply:
Do you prefer investigating it now or should we put it on a 'Performance ToDo List'/create a ticket?